deluge/core/torrentmanager.py

```diff
@@ -1213,11 +1267,49 @@ def torrents_status_update(self, torrent_ids, keys, diff=False):
         """
         d = Deferred()
         now = time.time()
+
+        session_id = component.get("RPCServer").get_session_id()
+        if session_id not in self.status_dicts:
+            self.status_dicts[session_id] = {"torrents_status": {}, "keys": {}}
+
+        if len(self.torrents_status_requests):
+            count = 0
+            for request in self.torrents_status_requests:
+                if request[0] == session_id:
+                    count += 1
+
+        if len(self.torrents_status_requests):
+            if len(self.torrents_status_requests) > 20:
+                log.warn("Too many status updates in queue - server overloaded?")
+
+                def return_empty():
+                    d.callback(({}, []))
+                reactor.callLater(0, return_empty)
+                return d
         # If last update was recent, use cached data instead of request updates from libtorrent
-        if (now - self.last_state_update_alert_ts) < 1.5:
-            reactor.callLater(0, self.handle_torrents_status_callback, (d, torrent_ids, keys, diff))
+        if (now - self.last_state_update_alert_ts) < 0.5 and False:
+            reactor.callLater(0, self.handle_torrents_status_callback, (session_id, d, torrent_ids,
+                                                                        keys, only_updated))
         else:
             # Ask libtorrent for status update
-            self.torrents_status_requests.insert(0, (d, torrent_ids, keys, diff))
+            self.torrents_status_requests.insert(0, (session_id, d, torrent_ids, keys, only_updated))
             self.session.post_torrent_updates()
         return d
+
+    def on_client_disconnected(self, session_id):
+        """
+        Clean up any torrents status requests stored for this session id
+
+        """
+        i = 0
+        removed = 0
+        while i < len(self.torrents_status_requests):
```

Comment:
for loop won't work as `self.torrents_status_requests` is changed inside the loop. The while is needed to avoid incrementing `i` when an element is removed. `orig_len - new_len` is a good point.